functionaltests/src/test/java/util/WebDriverSofortueberweisung.java

```diff
@@ -46,28 +48,35 @@ public WebDriverSofortueberweisung() {
             //swallow these messages
         });
         webClient.setCssErrorHandler(new SilentCssErrorHandler());
+
+        this.wait = new WebDriverWait(this, 20);
     }
 
     private void doLogin(String userid, String pin) {
         final WebElement useridInput = findElement(By.id(SU_LOGIN_NAME_ID));
-        final WebElement pinInput = findElement(By.id(SU_USER_PIN_ID));
+
+        wait.until(ExpectedConditions.elementToBeClickable(useridInput));
```

Comment:
How long it will wait? You wrote in the comment that they are enabled asynchronously when you change other inputs values. So these values are changed in different method?

Reply:
[it will be waiting up to 20 seconds](https://github.com/commercetools/commercetools-payone-integration/pull/180/files/776861acaa0c146bfed1b2b311e3d7c3a369f809#diff-4887a49c8d02ab752b32e97b6c562762R52)

> So these values are changed in different method?
we don't know how this values are changed. It depends on actual Sofortüberweisung front-end logic (JavaScript/HTML)

Reply:
So you don't fill in those values in your test?

Reply:
Yes, i fill all the values in the the same method 3 lines below